src/main/java/org/medicmobile/webapp/mobile/OpenSettingsDialogFragment.java

```diff
@@ -33,8 +37,44 @@ public boolean onTouch(View view, MotionEvent event) {
 	@Override
 	public void onCreate(@Nullable Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
+		setRetainInstance(true);
```

Comment:
So, I am probably missing something here, but in my testing, adding this line here in the `onCreate` was all that I needed to avoid the error in question.  

In my debugging it seemed like without this line, the `OpenSettingsDialogFragment.onCreate` method was getting called a bunch of times when entering/exiting split-screen mode _before the `EmbeddedBrowserActivity.onCreate` method is even called_.  When this happened, the `wbvMain` view was still null (presumably because the activity was not created yet).  Eventually, the `EmbeddedBrowserActivity.onCreate` method is called for the new activity and that triggers the  `OpenSettingsDialogFragment.onCreate` method to be called again (and this time the `wbvMain` view exists.  

_However_, if I set `setRetainInstance(true);` here in `OpenSettingsDialogFragment.onCreate`  and leave everything else as it is in `master`, it prevents all the extra initial calls to  `EmbeddedBrowserActivity.onCreate` that were happening before `EmbeddedBrowserActivity.onCreate` and so avoids the error.  When the `EmbeddedBrowserActivity.onCreate`  function is called, that triggers a call to `OpenSettingsDialogFragment.onCreate` which then can access the `wbvMain` view at that point in the lifecycle.  

I am not 100% sure about exactly what is happening here. The  `EmbeddedBrowserActivity.onCreate` is not getting called multiple times. My theory is that there is some weird connection between the activity lifecycle and the fragment lifecycle that is causing the `OpenSettingsDialogFragment.onCreate` function to get called at other points during the activity lifecycle as the original `EmbeddedBrowserActivity` is destroyed.  Setting `setRetainInstance` seems to decouple the fragment from the activity lifecycle so that this does not happen.

Reply:
That's a strange one. I tried this one as well. My Android studio was crashing with the debug option. \o/. So, I did the classic debugging using logging. I did not get the `EmbeddedBrowserActivity.onCreate`  being called multiple times or `EmbeddedBrowserActivity.onCreate` not being called and only `OpenSettingsDialogFragment.onCreate` being called. I've included screenshots of this below.

Nevertheless, just adding the `setRetainInstance(true);` to line 38 on the master branch for ` OpenSettingsDialogFragment` class did work. (Tested on virtual and physical phones).

Setting the `setRetainInstance(true);` does detach the fragment from its activity which is perfect for this use case. [Docs](https://developer.android.com/reference/android/app/Fragment#setRetainInstance(boolean)).

I'll refactor the changes to include just the `setRetainInstance(true);` line in the `OpenSettingsDialogFragment`. I'll do this in a different branch and send another PR if this change is ever required in the future.

![Screenshot_20241021_160923](https://github.com/user-attachments/assets/115ccf5f-636d-4a5a-8643-73ab75d587b1)

![Screenshot_20241021_161001](https://github.com/user-attachments/assets/2c8a8064-102e-4465-b834-13bace766982)